File: Glacier/BinaryDeployment/BinaryDeployment.lua
local fs = self:findFriends()
for i = 1, #fs do
  local b = fs[i].deployment
  if fs[i].type == "paladin" then
    for j = 1, 8 do
      if b % 2 == 0 then
        self:summon("soldier")
      else
        self:summon("archer")
      end
      local ns = self:findFriends()
      self:command(ns[#ns], "move", {x=66-6*j, y=fs[i].pos.y})
      b = math.floor(b/2)
    end
  else
    for j = 1, 8 do
      if b % 3 == 0 then
        self:summon("soldier")
      elseif b % 3 == 1 then
        self:summon("archer")
      else
        self:summon("griffin-rider")
      end
      local ns = self:findFriends()
      self:command(ns[#ns], "move", {x=66-6*j, y=fs[i].pos.y})
      b = math.floor(b/3)
    end
  end
end

File: Glacier/BinaryDeployment/BinaryDeployment.py
units = ['soldier', 'archer', 'griffin-rider']

for i in range(len(hero.findFriends())):
    result = ""
    unit = hero.findFriends()[i]
    if unit.type == "paladin":
        divide = 2
    else:
        divide = 3
    number = unit.deployment
    while number >= divide:
        a = number % divide
        number = (number - a) / divide
        result = a + result
    result = number + result

    while len(result) < 8:
        result = '0' + result

    for k in range(len(result)):
        m = result[k]
        hero.summon(units[m])
        hero.command(hero.built[len(hero.built) - 1], "move", {'x': 14 + k * 6, 'y': unit.pos.y})